docs: Update links in documentation and fix spelling mistakes (#3238)
timber/timber: docs/v2/guides/posts.md

## Extending `Timber\Post`

> If you need additional functionality that the `Timber\Post` class doesn't provide or if you want to have cleaner Twig templates, you can [extend the `Timber\Post` class](/docs/v2/guides/extending-timber/) with your own classes:

Reviewer: ```suggestion
If you need additional functionality that the `Timber\Post` class doesn’t provide or if you want to have cleaner Twig templates, you can [extend the `Timber\Post` class](/docs/v2/guides/extending-timber/) with your own classes:
```

$wp_posts = fancy_plugin_get_custom_posts(); // -> an array of WP_Posts
```

> In this scenario, you can still easily map each of these posts to instances of `Timber\Post` or the appropriate subclass, according to the [Class Map](./class-maps/#the-post-class-map) you've defined:

Reviewer: ```suggestion
In this scenario, you can still easily map each of these posts to instances of `Timber\Post` or the appropriate subclass, according to the [Class Map](./class-maps/#the-post-class-map) you’ve defined:
```

## Using posts or post collections in the context

> Timber will automatically set the `post` or `posts` variable for you in the context depending on the template file you're using. Read more about this in the [Context Guide](https://timber.github.io/docs/v2/guides/context/#template-contexts).

Reviewer: ```suggestion
Timber will automatically set the `post` or `posts` variable for you in the context depending on the template file you’re using. Read more about this in the [Context Guide](https://timber.github.io/docs/v2/guides/context/#template-contexts).
```
